GenerateSacrificeSettings.sh

```diff
@@ -0,0 +1,440 @@
+while [[ ! $sound_volume =~ ^[0-5]\.[0-9]+$ && ! $sound_volume =~ ^[sS]$ ]]; do
+    read -p "Enter a floating-point number between 0.0 and 5.0 to be used as factor for sound volume (or 's' for skip and use default): " sound_volume
+    if [[ ! $sound_volume =~ ^[0-5]\.[0-9]+$ && ! $sound_volume =~ ^[sS]$ ]]; then
+        print_error_message
+    fi
+done
+if [[ $sound_volume =~ ^[sS]$ ]]; then
+    echo "# --sound-volume=0.5                           # additional factor on sound volume" >> settings.txt
+else
+    echo "--sound-volume=$sound_volume                           # additional factor on sound volume" >> settings.txt
+fi
+
+echo # --no-advisor-help-speech                   # disable zyzyx >> settings.txt
+# end of sound options
+
+# Multiplayer part
+echo "### multiplayer options for host ###" >> settings.txt
+
+echo "1. Host for 1v1"
+echo "2. Host for 2v2"
+echo "3. Host for 3v3"
+echo "4. Join"
+echo "5. Host FFA"
+
+hostmode=""
+while [[ ! $hostmode =~ ^[1-5]$ ]]; do
+    read -p "Enter a number between 1 and 5: " hostmode
+    if [[ ! $hostmode =~ ^[1-5]$ ]]; then
+        print_error_message
+    fi
+done
+
+## HOSTMODE
+case $hostmode in
+    # FFA
+    5)  
+		echo "1. Host for 3"
+		echo "2. Host for 4"
+		echo "3. Host for 5"
+		echo "Hosting FFA for 6 is not available due to lack of maps and maplists"
+		ffamode=""
+		while [[ ! $ffamode =~ ^[1-3]$ ]]; do
+			read -p "Enter a number between 1 and 3: " ffamode
+			if [[ ! $ffamode =~ ^[1-3]$ ]]; then
+				print_error_message
+			fi
+		done
+		ffaplayers=""
+		case $ffamode in
+			1) ffaplayers=3 ;;
+			2) ffaplayers=4 ;;
+			3) ffaplayers=5 ;;
+		esac
+
+		# Finally, echo everything to settings.txt
+		echo "--host=$ffaplayers" >> settings.txt
+		echo "--ffa                                        # host ffa (overrides team settings)" >> settings.txt
+		echo "--random-gods" >> settings.txt
+		echo "--shuffle-sides" >> settings.txt
+		echo "--stutter-on-desynch" >> settings.txt
+		echo "--no-particles" >> settings.txt
+		# if there's only 3 ffa players, offer another choice, otherwise use tg-{$ffaplayers}ffa.txt maplist
+		if [ $ffaplayers != 3 ]; then
+			echo "--map-list=maps-tg-{ffaplayers}ffa.txt" >> settings.txt
+		else
+			# another maplist choice
+			echo "1. shiny's 3ffa map list"
+			echo "2. tg's 3ffa map list"
+			echo "3. tree's 3ffa map list"
+			maplist=""
+			while [[ ! $maplist =~ ^[1-3]$ ]]; do
+				read -p "Enter a number between 1 and 3: " maplist
+				if [[ ! $maplist =~ ^[1-3]$ ]]; then
+					print_error_message
+				fi
+			done
+			case $maplist in
+				1) # shiny ffa3
+					echo "--map-list=maps-shiny-3ffa.txt" >> settings.txt
+					;;
+				2) # tg ffa3
+					echo "--map-list=maps-tg-3ffa.txt" >> settings.txt
+					;;
+				3) # tree ffa3
+					echo "--map-list=maps-tree-3ffa.txt" >> settings.txt
+					;;
+			esac
+		fi
+		;;
+    # join
+    4)
+		echo "--join" >> settings.txt
+		# call the observer() function we declared at the bottom of the file
```

Comment:
comment no longer correct since function definitions had to be moved before any calls to them were made... please fix.